src/radio/Radio.tsx

```diff
@@ -1,92 +1,38 @@
 import React from "react";
-import {
-  Radio as ReakitRadio,
-  RadioProps as ReakitRadioProps,
-  RadioGroup as ReakitRadioGroup,
-  RadioGroupProps as ReakitRadioGroupProps,
-} from "reakit";
+import { Radio as ReakitRadio, RadioProps as ReakitRadioProps } from "reakit";
 import { cx } from "@renderlesskit/react";
 
 import { useTheme } from "../theme";
-import { BoxProps, Box } from "../box";
-import { createContext } from "../utils";
+import { RadioIcon } from "./RadioIcon";
+import { useRadioContext } from "./RadioGroup";
 import { forwardRefWithAs } from "../utils/types";
-import {
-  useRadioState,
-  RadioInitialState,
-  RadioStateReturn,
-} from "./useRadioState";
 
-const [RadioProvider, useRadioContext] = createContext<{
-  radioState: RadioStateReturn;
-  radioSize?: RadioLabelProps["size"];
-}>({
-  name: "RadioContext",
-  strict: true,
-  errorMessage: "Radio must be used within RadioContextProvider",
-});
-
-export interface RadioLabelProps extends BoxProps {
-  /**
-   * How large should the radio be?
-   */
-  size?: keyof Renderlesskit.GetThemeValue<"radio", "size">;
-}
-
-export const RadioLabel = forwardRefWithAs<
-  RadioLabelProps,
-  HTMLLabelElement,
-  "label"
->(({ children, className, size, ...props }, ref) => {
-  const theme = useTheme();
-  const { radioSize } = useRadioContext();
-  const _size = size || radioSize || "sm";
-
-  return (
-    <Box
-      as="label"
-      ref={ref}
-      className={cx(theme.radio.base, theme.radio.size[_size], className)}
-      {...props}
-    >
-      {children}
-    </Box>
-  );
-});
+export type RadioCommonProps = {
+  disabled?: boolean;
+  value?: ReakitRadioProps["value"];
+  size?: keyof Renderlesskit.GetThemeValue<"radio", "icon">["size"];
+};
 
 export const Radio = forwardRefWithAs<
   Partial<ReakitRadioProps>,
```

Comment:
```suggestion
  ReakitRadioProps
```

Reply:
Had to do `RadioHTMLProps & RadioCommonProps` since ReakitRadioProps also expects all the composite props.

Reply:
Right !